<?xml version="1.0" encoding="UTF-8"?>
<configurations>
  <UVApp id="vrsim"> 
  
   <!-- NavCon parameters -->
   <param name="Ts" value="0.1" />
  
   	<!-- dH controller parameters -->
    <param name="w_h" value="0.7" />
    <param name="attack_angle" value="45" />
  
    <!-- dV controller parameters -->
    <param name="w_v" value="0.3" />
    <param name="Xi" value="0" />
  
    <!-- Yaw controller parameters -->
    <param name="w_r" value="0.7" />
  
    <!-- Depth controller parameters -->
    <param name="w_w" value="0.3" />
    
    <!-- Yaw model parameters --> 
    <param name="alpha_r" value="3.523" />
    <param name="beta_r" value="1.49" />
    <param name="Max_N" value="3.5" />
    
    <!-- Depth model parameters -->  
    <param name="alpha_w" value="25.4857" />
    <param name="beta_w" value="11.1815" />
    <param name="Max_Z" value="3" />
    
    <line T1="[3](0,0,0)" T2="[3](1,1,0)" />
  	<!-- measurement vector <z,psi,dV,dH> -->
  	<param name="R" value="diag([4](0.1,0.02,1,1))" />
  	<param name="V" value="diag([4](1,1,1,1))" />
  	<!-- noise vector <u,w,r,b,vertCurrent,horzCurrent> -->
  	<param name="Q" value="diag([6](0.1,0.1,0.1,0.1,0.1,0.1))" />
  	<param name="W" value="[10,6]((1,0,0,0,0,0),
				(0,1,0,0,0,0),
				(0,0,1,0,0,0),
				(0,0,0,0,0,0),
				(0,0,0,0,0,0),
 				(0,0,0,0,0,0),
 				(0,0,0,0,0,0),
				(0,0,0,1,0,0),
				(0,0,0,0,1,0),
				(0,0,0,0,0,1))" />  
	<!-- filter options -->
	<param name="initial-state" value="[10](0,0,0,0,0,0,0,0,0,0)" />
  	<param name="sampling-time" value="0.1" /> 
  
  
  
  
  
  
   <!-- Timeout value -->
   <watchdog timeout="10000" loop="1000" />
   <!-- Plugin configuration -->
   <plugin name="vrsim-plug" id="falcon"/>
   <!-- open loop surge force during LF -->
   <OpenLoopTauX>10</OpenLoopTauX>
   
   <saveIdentification>1</saveIdentification>
   
   <!-- Identification parameters -->
   <heading-so tau="2" angle="15" />
   <depth-so tau="2" depth="0.5" />
   
   <!-- The environment initialization -->
   <!-- <current-force>[3](10,0,0)</current-force> -->
      
   <!-- Vehicle simulation parameters --> 
   <VehicleModel id="falcon">
    <sampling-time>0.1</sampling-time>
    <mass>20</mass>
    <gravity>10</gravity>
    <inertia-matrix>diag([3](30,30,3.523))</inertia-matrix>
    <rg>[3](0,0,0)</rg>
    <rb>[3](0,0,0)</rb>
    <rho>1025</rho>
    <lift-force>200</lift-force>
    <added-mass>diag([6](20,80,5.4857,15,30,0))</added-mass>
    <linear-damping>diag([6](100,100,11.1815,400,300,1.49))</linear-damping>
    <quadratic-damping>diag([6](0,0,0,0,0,0))</quadratic-damping>
    <eta0>[6](10,10,0,0,0,0)</eta0>
    <nu0>[6](0,0,0,0,0,0)</nu0>
    <coupled>0</coupled>
    
    <current>[3](0,0,0)</current>
    
    <!-- <process-noise>[6](0.1,0.1,0.1,0.1,0.1,0.1)</process-noise> -->
    <!-- <measurement-noise>[6](1,1,0.1,0.01,0.01,0.01)</measurement-noise> -->
   </VehicleModel> 
   
   
  </UVApp>
</configurations>
